web/app/components/new/doc-form.ts

```diff
@@ -0,0 +1,242 @@
+    } else {
+      this.formRequirementsMet = false;
+    }
+    if (this.validateEagerly) {
+      this.validate();
+    }
+  }
+
+  /**
+   * Validates the form and updates the `formErrors` property.
+   */
+  private validate() {
+    const errors = { ...FORM_ERRORS };
+    if (this.productAbbreviation) {
+      if (/\d/.test(this.productAbbreviation)) {
+        errors.productAbbreviation =
+          "Product abbreviation can't include a number";
+      }
+    }
+    this.formErrors = errors;
+  }
+
+  /**
+   * Returns contributor emails as an array of strings.
+   */
+  private getEmails(values: HermesUser[]) {
+    return values.map((person) => person.email);
+  }
+
+  @action protected registerForm(form: HTMLFormElement) {
+    this._form = form;
+  }
+
+  /**
+   * Binds the FormData to our locally tracked properties.
+   * If the summary is long, shows a gentle warning.
+   * Conditionally validates.
+   */
+  @action protected updateForm() {
+    const formObject = Object.fromEntries(new FormData(this.form).entries());
+
+    assert("title is missing from formObject", "title" in formObject);
+    assert("summary is missing from formObject", "summary" in formObject);
+
+    this.title = formObject["title"] as string;
+    this.summary = formObject["summary"] as string;
+
+    if ("productArea" in formObject) {
+      this.productArea = formObject["productArea"] as string;
+    }
+
+    if (this.summary.length > 200) {
+      this.summaryIsLong = true;
+    } else {
+      this.summaryIsLong = false;
+    }
+
+    this.maybeValidate();
+  }
+
+  /**
+   * Updates the contributors property and conditionally validates the form.
+   */
+  @action protected updateContributors(contributors: HermesUser[]) {
+    this.contributors = contributors;
+  }
+
+  /**
+   * Validates the form, and, if valid, creates a document.
+   * If the form is invalid, sets `validateEagerly` true.
+   */
+  @action protected submit(event: SubmitEvent) {
+    event.preventDefault();
+    this.validateEagerly = true;
+    this.validate();
+    if (this.formRequirementsMet && !this.hasErrors) {
+      this.createDoc.perform();
+    }
+  }
+
+  /**
+   * Creates a document draft, then redirects to the document.
+   * On error, show a flashMessage and allow users to try again.
+   */
+  private createDoc = task(async () => {
+    this.docIsBeingCreated = true;
+
+    try {
+      const doc = await this.fetchSvc.fetch("/api/v1/drafts", {
+        method: "POST",
+        headers: { "Content-Type": "application/json" },
+        body: JSON.stringify({
+          contributors: this.getEmails(this.contributors),
+          docType: this.args.docType,
+          owner: this.authenticatedUser.info.email,
+          product: this.productArea,
+          productAbbreviation: this.productAbbreviation,
+          summary: cleanString(this.summary),
+          title: cleanString(this.title),
+        }),
+      });
+
+      throw new Error();
```

Comment:
Was this for testing and should be removed?

Reply:
😅😬🫠 thanks for catching